tasks/process_agent.py

```diff
@@ -95,3 +96,33 @@ def build(ctx, race=False, go_version=None, incremental_build=False, puppy=False
     }
 
     ctx.run(cmd.format(**args), env=env)
+
+
+@task
+def build_dev_image(ctx, image=None, push=False, base_image="datadog/agent:latest"):
+    """
+    Build a dev image of the process-agent based off an existing datadog-agent image
+
+    image: the image name used to tag the image
+    push: if true, run a docker push on the image
+    base_image: base the docker image off this already build image (default: datadog/agent:latest)
+    """
+    if image == None:
+        raise Exit(message="image was not specified")
+
+    docker_context = tempfile.mkdtemp()
```

Comment:
I wondering if you should automatically cleanup the `docker_context` folder? 

looking at the [doc](https://docs.python.org/3/library/tempfile.html) you should use `TemporaryDirectory` for automatic cleanup, else with `mkdtemp()` you should do the cleanup yourself.

> mkstemp() and mkdtemp() are lower-level functions which require manual cleanup.

Reply:
Nice catch, `TemporaryDirectory` was added in python3, and pyinvoke seems to use 2.7 on our dev VM. I created a TempDir() class that can be used with a `with` statement.